x/stakeibc/keeper/deposit_records.go

```diff
@@ -47,8 +47,8 @@ func (k Keeper) TransferExistingDepositsToHostZones(ctx sdk.Context, epochNumber
 		pstr := fmt.Sprintf("\t[TransferExistingDepositsToHostZones] Processing deposits {%d} {%s} {%d}", depositRecord.Id, depositRecord.Denom, depositRecord.Amount)
 		k.Logger(ctx).Info(pstr)
 
-		// if a TRANSFER_QUEUE record has 0 balance and was created in the previous epoch, it's safe to remove since it will never be updated or used"
-		if depositRecord.Amount <= 0 {
+		// if a TRANSFER_QUEUE record has 0 balance and was created in the previous epoch, it's safe to remove since it will never be updated or used
+		if depositRecord.Amount <= 0 && depositRecord.DepositEpochNumber < epochNumber {
```

Comment:
I think you can remove this extra condition - it's accounted for [here](https://github.com/Stride-Labs/stride/pull/296/commits/eff92d824f21ff19e54f81d03481f5d2b9075169#diff-8edb9f84b67f6d1f48651b601161499f94253b8fce28ce0a405b9d2f50e19dd6R39)

Reply:
I was thinking about this, felt slightly safer to add the check in case we change the above code. Might not be needed though, wdyt?

Reply:
imo, we should just remove it. Considering it's a few lines above, I don't think there's a need to code that defensively. 

If it was hidden in another file, then I could see the case for the additional safety check